src/main/java/it/pagopa/pagopa/apiconfig/service/CdiService.java

```diff
@@ -161,6 +172,62 @@ public void deleteCdi(String idCdi, String pspCode) {
         cdiMasterRepository.delete(cdiMaster);
     }
 
+
+    public void uploadHistory() {
+        var result = new ArrayList<CdiCosmos>();
+        cdiMasterValidRepository.findAll()
+                .stream()
+                .filter(Objects::nonNull)
+                .forEach(master -> {
+                    var cdiDetails = master.getCdiDetail() != null ?
+                            master.getCdiDetail()
+                                    .stream()
+                                    .filter(Objects::nonNull)
+                                    .map(this::mapDetails)
+                                    .collect(Collectors.toList())
+                            : null;
+                    result.add(CdiCosmos.builder()
+                            .idPsp(master.getFkPsp().getIdPsp())
+                            .idCDI(master.getIdInformativaPsp())
+                            .cdiStatus("NEW")
+                            .digitalStamp(master.getMarcaBolloDigitale())
+                            .validityDateFrom(master.getDataInizioValidita() != null ? master.getDataInizioValidita().toLocalDateTime() : null)
+                            .details(cdiDetails)
```

Comment:
and if `cdiDetails` is null, how the flow is managed? could we skip `result.add`?

Reply:
can I throw an exception?

Reply:
I think it is better...